src/blame.rs

```diff
@@ -0,0 +1,512 @@
+  pub fn is_empty(&self) -> bool {
+    self.inner.is_empty()
+  }
+
+  /// Returns an iterator over the hunks in this blame.
+  ///
+  /// This internal method is used to implement getHunks() and is not
+  /// directly exposed to JavaScript.
+  pub fn iter(&self) -> BlameIter {
+    BlameIter {
+      blame: self,
+      idx: 0,
+      len: self.get_hunk_count() as usize,
+    }
+  }
+
+  #[napi]
+  /// Generates blame information from an in-memory buffer
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   buffer(buffer: Buffer, bufferLen: number): Blame;
+  /// }
+  /// ```
+  ///
+  /// @example
+  /// ```ts
+  /// const buffer = Buffer.from('modified content');
+  /// const bufferBlame = blame.buffer(buffer, buffer.length);
+  /// ```
+  ///
+  /// @param {Buffer} buffer - Buffer containing file content to blame
+  /// @param {number} buffer_len - Length of the buffer in bytes
+  /// @returns A new Blame object for the buffer content
+  /// @throws If the buffer contains invalid UTF-8
+  pub fn buffer(&self, buffer: Buffer, buffer_len: u32, env: Env) -> Result<Blame> {
+    let content = std::str::from_utf8(&buffer[..buffer_len as usize])
+      .map_err(|e| Error::from_reason(format!("Invalid UTF-8 in buffer: {}", e)))?;
+
+    let blame = match &self.inner {
+      BlameInner::Repo(shared_ref) => {
+        let cloned = shared_ref.clone(env)?;
+
+        cloned.share_with(env, |git_blame| {
+          git_blame
+            .blame_buffer(content.as_bytes())
+            .map_err(|e| Error::from(crate::Error::from(e)))
+        })?
+      }
+    };
+
+    Ok(Blame {
+      inner: BlameInner::Repo(blame),
+    })
+  }
+
+  #[napi]
+  /// Gets blame information for the specified index
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   getHunkByIndex(index: number): BlameHunk;
+  /// }
+  /// ```
+  ///
+  /// @param {number} index - Index of the hunk to get (0-based)
+  /// @returns Blame information for the specified index
+  /// @throws If no hunk is found at the index
+  pub fn get_hunk_by_index(&self, index: u32) -> Result<BlameHunk> {
+    let hunk = self
+      .inner
+      .get_index(index as usize)
+      .ok_or_else(|| Error::from_reason(format!("No blame hunk found at index {}", index)))?;
+
+    let commit_id = hunk.final_commit_id().to_string();
+    let is_zero_commit = commit_id == "0000000000000000000000000000000000000000";
+
+    let signature = if is_zero_commit {
+      None
+    } else {
+      Signature::try_from(hunk.final_signature()).ok()
+    };
+
+    let path = hunk.path().map(|p| p.to_string_lossy().to_string());
+
+    Ok(BlameHunk {
+      commit_id,
+      final_start_line_number: hunk.final_start_line() as u32,
+      lines_in_hunk: hunk.lines_in_hunk() as u32,
+      signature,
+      path,
+      orig_start_line_number: hunk.orig_start_line() as u32,
+      is_boundary: hunk.is_boundary(),
+    })
```

Comment:
This implementation is duplicated. How about implementing `impl<'a> From<&git2::BlameHunk<'a>> for BlameHunk`?